src/ServiceFabric/ServiceFabric/Commands/UpdateAzServiceFabricVmImage.cs

```diff
@@ -0,0 +1,74 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.ServiceFabric.Commands
+{
+    using System.Management.Automation;
+    using Microsoft.Azure.Commands.ServiceFabric.Models;
+    using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+    [Cmdlet("Update", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ServiceFabricVmImage", SupportsShouldProcess = true), OutputType(typeof(PSCluster))]
+    public class UpdateAzServiceFabricVmImage : ServiceFabricClusterCmdlet
+    {
+        /// <summary>
+        /// Resource group name
+        /// </summary>
+        [Parameter(Mandatory = true, Position = 0, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Specify the name of the resource group.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty()]
+        public override string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, Position = 1, ValueFromPipelineByPropertyName = true,
+                   HelpMessage = "Specify the name of the cluster")]
+        [ValidateNotNullOrEmpty()]
+        [Alias("ClusterName")]
+        public override string Name { get; set; }
+
+        /// <summary>
+        /// VmImage sets the Service Fabric runtime package delivery hint for the cluster, which results in dropping the appropriately-mapped package for the VM image OS.
+        /// Accepted values: Windows, Linux, Ubuntu, Ubuntu18_04
+        /// Note: Linux & Ubuntu both map to Ubuntu 16.04
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true,
+                   HelpMessage = "VM Image hint for determining SF runtime package delivery: i.e. .cab/.deb")]
+        [ValidateNotNullOrEmpty()]
+        public VmImageKind VmImage { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            var cluster = GetCurrentCluster();
+            var oldVmImage = GetVmImage(cluster);
```

Comment:
We shouldn't allow going from windows to Linux and vise versa right? should we add the validation here?

Reply:
If there is investment in hybrid clusters this setting would expectedly get deprecated, or another "Hybrid" or similar setting would be introduced to allow such combinations. In theory a Linux to Windows switch would be fine if the running apps continue to work.

As long as extension does 'PUT node' with OS details and uses the bootstrap API for retrieving cluster package, this setting will only really used to deliver packages for upgrades, so this would be the earnest indicator that the user intends to switch the primary OS of the cluster to Windows and future incoming packages for upgrades will be based on the cab.

The scenario could be as such:
- User has an app that runs dotnet core which is assumed to be able to activate on Windows & Linux machines without repackaging.
- vmImage is updated from Ubuntu18_04 to Windows so upgrades will not work until all cluster nodes are on a homogeneous OS
- A new node type is created with a WindowsServer OS and the app is deployed to the nodes as they come up
- The old node type is decommissioned